src/main/java/org/spongepowered/api/command/args/GenericArguments.java

```diff
@@ -1584,27 +1609,57 @@ public static CommandElement entity(Text key) {
      * @return The command element
      */
     public static CommandElement entityOrSource(Text key) {
-        return new EntityCommandElement(key, true);
+        return new EntityCommandElement(key, true, false, null);
+    }
+
+    public static CommandElement entity(Text key, Class<? extends Entity> type) {
```

Comment:
Probably should document this and the two methods below(`entity(Text key, Class<? extends Entity> type)`, `entityOrTarget(Text key)`, and `entityOrTarget(Text key, Class<? extends Entity> type)` , explaining what they accomplish.

As a whole this PR seems good and I would love to see it in.